### DISCUSSION TITLE
Powerwall not found

### DISCUSSION THREAD

**Discussion by shawwellpete** [alandtse/tesla#779] [Q&A]
Firstly thank you for a fantastic integration which I have been using for the two Teslas in the family for a while. 

I found out that the integration also works with powerwalls but mine is not found when I install the integration, the only devices discovered are the two cars with no mention of the Tesla gateway or the powerwalls.

I have tried removing the integration and installing again, and also removed the official powerwall integration in case that caused a conflict. 

Any other ideas?

**Comment by shawwellpete** [comment]
Any suggestions please?

**Comment by purcell-lab** [comment]
Goto Developer Tools / Services and run the Tesla API with the PRODUCT_LIST endpoint and you can see what is attached to your account:

![image](https://github.com/alandtse/tesla/assets/79175134/c34b1d88-b5ad-4a42-b246-26d3c2f8f700)

**Reply by shawwellpete** [reply]
Thank you for taking the time to help Mark, much appreciated.
I think I have the same parameters as your screenshot but get an error...
![image](https://github.com/alandtse/tesla/assets/74064242/b4ca0203-4df3-4d8f-a712-fb69f241f5f6)

**Reply by skipishere** [reply]
I tried the above and founds if the "1" has quotes it failed with Unknown Error. Removing it worked.
For simplicity here's the YAML mode version you can run there instead.

```YAML
service: tesla_custom.api
data:
  parameters:
    path_vars:
      vehicle_id: 1
    wake_if_asleep: true
  command: PRODUCT_LIST
```

For me, I could see my car and then after that my powerwall

**Reply by shawwellpete** [reply]
Thanks Stephen,

Odd...

![image](https://github.com/alandtse/tesla/assets/74064242/68740c14-8bc1-4a8d-9c29-bd7d0124d093)

**Reply by grillp** [reply]
I have the same issue. I Only see 1 car and no PowerWall entities.

When I run the command above I see my car and my energy site information, with `resource_type: battery', in the result.

I only got my powerWall a few weeks ago, long after I had by EV and have had his integration running for may years.

Do I need to somehow 'refresh' the configuration? If I uninstall and reinstall will it add then add the battery?

**Reply by grillp** [reply]
I just installed the integration again on a dev HA instance and when I activated it, it gave me the option of adding my Energy Products as well as my car.. With all battery sensors there.
I then now deleted the integration in my production server and re-added it, when it asked for my Tesla creds again and it found the battery too. All my car sensors were restored. Note that I had not changed any of their names, so they seamlessly slotted back into all my automations that used them. Even where I used Device conditions or settings. 

I think it could even be possible to just add a 'new' integration with the same username and just select the Energy Products on the dialog where you add your creds. I have not tried that but it could work 🤷

**Comment by shawwellpete** [comment]
Thanks for the suggestions so far, I am still htting a brick wall with this. 
Following @grillp 's suggestion I installed a blank HA instance and installed HACS and the Tesla custom integration.

Again, only the two cars come up, however the service call suggestion from @purcell-lab  does work (Onloy shows infor for the cars though). 

![image](https://github.com/alandtse/tesla/assets/74064242/2e87b37d-735a-4408-ac48-51f1b72da955)

running the service on my normal HA still gives an error. 

![image](https://github.com/alandtse/tesla/assets/74064242/4bc9c037-7108-4a83-999e-ce2cdf53b0f5)

Logging into tesla.com I do not see the powerwall? It is on my app though. Do you guys see the powerwall on your account?

![image](https://github.com/alandtse/tesla/assets/74064242/bc8d1178-1b92-4ccd-9ea8-533685940408)

**Reply by skipishere** [reply]
Powerwalls don't appear on that page (at least for me).

**Comment by shawwellpete** [comment]
Hi guys, I would still appreciate help or suggestions on this

**Reply by skipishere** [reply]
Really not sure what else we can suggest. Assuming your powerwall appears in your Tesla app with your cars and you use the same account for home assistant then it should work. 🤷

**Reply by shawwellpete** [reply]
I'll start by correcting my typing in my request for help! It's a very odd one. I don't think I am doing anything wrong.

![2024-01-13_12-19-04](https://github.com/alandtse/tesla/assets/74064242/d565f738-5b3b-4aca-807c-698800aa3a5a)
![2024-01-13_12-19-55](https://github.com/alandtse/tesla/assets/74064242/a13a528c-8391-43b6-b99d-47996fc00794)

Very odd.

**Reply by skipishere** [reply]
Out of interest, have you tried doing the same but without the vehicles? Can't imagine it would make a difference but might be worth a shot

**Reply by shawwellpete** [reply]
Good idea, and one I tried 5 minutes ago! I wondered if the integration was limited to 2 devices so wouldn't work on 2 cars and a power wall. However if I exclude the cars, no devices are brought in.

**Comment by purcell-lab** [comment]
You can access a command line solution to explore what endpoints your Tesla account is returning.

https://github.com/zabuldon/teslajsonpy/issues/342

If the command line works for you then you can include those commands into Home Assistant.

**Reply by shawwellpete** [reply]
trying to work out how to do that, thanks

**Reply by shawwellpete** [reply]
> You can access a command line solution to explore what endpoints your Tesla account is returning.
> 
> [zabuldon/teslajsonpy#342](https://github.com/zabuldon/teslajsonpy/issues/342)
> 
> If the command line works for you then you can include those commands into Home Assistant.

I have a bit of time so I am coming back to this, I'm resilient if not successful!

I have been trying to get [this](https://github.com/zabuldon/teslajsonpy/issues/342) commands working but, having never used a python CLi am doing something stupid and, despite me searching, can't see what

I downloaded the python code from the git hub, unzipped it and in terminal on the mac went to the directory with 

`cd /Users/myname/Downloads/TeslaPy-cli-updates/teslapy`

I have python 3 installed, I can run it by typing 'python3' for example

However, when I try

`python3 -e me@email.com -z PRODUCT_LIST` obviosuly with my correct Tesla account email address I get the error

`Unknown option: -e
usage: /Library/Frameworks/Python.framework/Versions/3.12/Resources/Python.app/Contents/MacOS/Python [option] ... [-c cmd | -m mod | file | -] [arg] ...
Try `python -h' for more information.`

Any ideas?
